distributed/worker.py

```diff
@@ -332,6 +332,15 @@ def _register_with_scheduler(self):
                 middle = (_start + _end) / 2
                 self.scheduler_delay = response['time'] - middle
                 self.status = 'running'
+
+                # Retrieve eventual init functions and run them
+                for init_function_bytes in response['init-functions']:
+                    init_function = pickle.loads(init_function_bytes)
+                    if has_arg(init_function, 'dask_worker'):
+                        result = init_function(dask_worker=self)
+                    else:
+                        result = init_function()
+                    logger.info('Init function %s ran: output=%s' % (init_function, result))
```

Comment:
It looks like this code fails under the test `py.test distributed/tests/test_scheduler.py::test_worker_name`

Reply:
Yes thanks, I ran the init functions even if scheduler response was not OK, I've move the init code.